torch/csrc/jit/frontend/ir_emitter.cpp

```diff
@@ -3453,17 +3453,36 @@ struct to_ir {
       args.emplace_back(loc, "end", end);
     }
     if (sliceable->type()->cast<TupleType>()) {
-      if (step) {
-        // TODO: add support for slicing tuples with a step
-        throw ErrorReport(loc)
-            << "Unsupported operation: slicing tuples with a step isn't supported";
+      if (start && end && step) {
```

Comment:
I think logic in this method can be simplified.

You can change type of `args` to `std::vector<at::optional<NamedValue>>`, then construct its 4 values according to nullness of start/end/step. This way you only need to make one call to `emitTupleSlice`.

Reply:
Good idea! will do